packages/react/src/components/Modal/Modal.js

```diff
@@ -283,32 +284,38 @@ const Modal = React.forwardRef(function Modal(
   );
 
   return (
-    <div
-      {...rest}
-      onKeyDown={handleKeyDown}
-      onMouseDown={handleMousedown}
-      onBlur={handleBlur}
-      className={modalClasses}
-      role="presentation"
-      ref={ref}>
-      {/* Non-translatable: Focus-wrap code makes this `<span>` not actually read by screen readers */}
-      <span
-        ref={startTrap}
-        tabIndex="0"
-        role="link"
-        className={`${prefix}--visually-hidden`}>
-        Focus sentinel
-      </span>
-      {modalBody}
-      {/* Non-translatable: Focus-wrap code makes this `<span>` not actually read by screen readers */}
-      <span
-        ref={endTrap}
-        tabIndex="0"
-        role="link"
-        className={`${prefix}--visually-hidden`}>
-        Focus sentinel
-      </span>
-    </div>
+    <Portal>
```

Comment:
I think that rendering the modal in a portal by default would be a breaking change. The majority of usage in the wild probably already renders the modal via a portal and would result in nested portals. Users could/would run into not being able to access events of the innermost portal, https://github.com/facebook/react/issues/14540

Reply:
One way to evaluate backwards compatibility would be to leave the existing 'with state manager' story as-is and use it as a testing grounds for how most modals are used right now.

Reply:
good point